packages/bundlers/experimental/src/ExperimentalBundler.js

```diff
@@ -696,88 +870,282 @@ function createIdealGraph(
       reachableRoots,
     ).reverse();
 
+    let reachableEntries = [];
+    let reachableNonEntries = [];
+
+    // Filter out entries, since they can't have shared bundles.
+    // Neither can non-splittable, isolated, or needing of stable name bundles.
+    // Reserve those filtered out bundles since we add the asset back into them.
+    for (let a of reachable) {
+      if (
+        entries.has(a) ||
+        !a.isBundleSplittable ||
+        getBundleFromBundleRoot(a).needsStableName ||
```

Comment:
Are we sure this is necessary and not already covered by entries? I'm not sure if we want to conflate naming with bundling behavior?

Reply:
I think we took this logic directly from the default bundler (see this [link to the code](https://github.com/parcel-bundler/parcel/blob/v2/packages/bundlers/default/src/DefaultBundler.js#:~:text=let%20entries%20%3D%20b,b.needsStableName%20%26%26)) 

Yeah I'm kind of confused about this myself, I know non-entry bundles can be non-splittable and needStableName so that's why that is there, but I'm not sure if **only** bundlingBehavior can cover that. `NeedsStableName` is determined **by** bundlingBehavior but not equal, so they are related. As for `isBundleSplittable` , I believe that comes from the Asset itself, so I'm not sure how thats determined exactly but I think it's set before bundling occurs

Reply:
Oh hmm I don't remember why we never changed this. Ok, well we can come back to it.